portfolio/src/main/java/com/google/sps/servlets/BookServlet.java

```diff
@@ -0,0 +1,74 @@
+package com.google.sps.servlets;
+
+import com.google.gson.Gson;
+import com.google.sps.data.Book;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Map;
+import java.util.Scanner;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * Returns book titles and reviews as a JSON array, e.g. [{"title": Othello, "reviews": {"Nice",
+ * "Bad"}}]
+ */
+@WebServlet("/book-data")
+public class BookServlet extends HttpServlet {
+  private Collection<Book> books;
+
+  @Override
+  public void init() throws ServletException {
+    books = new ArrayList<>();
+    readBooks();
+  }
+
+  private void readBooks() throws ServletException {
```

Comment:
(feel free to do this in separate PR)
This logic could be moved to a separate Java class, and takes csv file path as constructor parameter. This will allow you to easily use this in non-servlet code as well, if needed. It would also be easy to write test cases (which is also encouraged when you make this change).

BookReader bookReader = new BookReader("./path/to/my/book.csv");

see: https://en.wikipedia.org/wiki/Single-responsibility_principle
books = bookReader.readAll();

You could also consider using http://opencsv.sourceforge.net/#how_can_i_use_it_in_my_maven_projects

Reply:
Ah! That makes sense! I'll implement this + add test cases in the next PR :)